test/e2e/plugin_lifecycle_test.go

```diff
@@ -0,0 +1,155 @@
+			},
+			Spec: greenhousev1alpha1.PluginSpec{
+				PluginDefinition: "nginx-18.1.7",
+				ReleaseNamespace: setup.Namespace(),
+				ClusterName:      secret.Name,
+				OptionValues: []greenhousev1alpha1.PluginOptionValue{
+					{
+						Name:  "replicaCount",
+						Value: &apiextensionsv1.JSON{Raw: []byte("1")},
+					},
+				},
+			},
+			Status: greenhousev1alpha1.PluginStatus{},
+		}
+
+		pluginDefinitionList := &greenhousev1alpha1.PluginDefinitionList{}
+		pluginList := &greenhousev1alpha1.PluginList{}
+		deploymentList := &appsv1.DeploymentList{}
+		ctx := test.Ctx
+
+		//Creating plugin definition
+		err := test.K8sClient.Create(ctx, testPluginDefinition)
+		Expect(err).NotTo(HaveOccurred())
+		err = test.K8sClient.List(ctx, pluginDefinitionList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginDefinitionList.Items)).To(BeEquivalentTo(1))
+
+		//Creating plugin
+		err = test.K8sClient.Create(ctx, testPlugin)
+		Expect(err).NotTo(HaveOccurred())
+		err = test.K8sClient.List(ctx, pluginList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginList.Items)).To(BeEquivalentTo(1))
+
+		//Checking deployment
+		err = remoteClient.List(ctx, deploymentList, client.InNamespace(setup.Namespace()))
+		Expect(err).NotTo(HaveOccurred())
+		SetDefaultEventuallyTimeout(60 * time.Second)
+		Eventually(func(g Gomega) bool {
+			err = remoteClient.List(ctx, deploymentList, client.InNamespace(setup.Namespace()))
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(len(deploymentList.Items) > 0).To(BeTrue())
+			return true
+		}).Should(BeTrue())
+
+		//Checking the name of deployment
+		nginxDeploymentExists := false
+		for _, deployment := range deploymentList.Items {
+			if strings.Contains(deployment.Name, "nginx") {
+				nginxDeploymentExists = true
+				Expect(deployment.Spec.Replicas).To(BeEquivalentTo(ptr[int32](1)))
+				break
+			}
+		}
+		Expect(nginxDeploymentExists).To(BeTrue())
+
+		//Updating replicas
+		err = test.K8sClient.List(ctx, pluginList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginList.Items)).To(BeEquivalentTo(1))
+		testPlugin = &pluginList.Items[0]
+		testPlugin.Spec.OptionValues[9].Value.Raw = []byte("2")
+		err = test.K8sClient.Update(ctx, testPlugin)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(func(g Gomega) bool {
+			err = remoteClient.List(ctx, deploymentList, client.InNamespace(setup.Namespace()))
+			g.Expect(err).NotTo(HaveOccurred())
+			for _, deployment := range deploymentList.Items {
+				if strings.Contains(deployment.Name, "nginx") {
+					g.Expect(deployment.Spec.Replicas).To(BeEquivalentTo(ptr[int32](2)))
+				}
+			}
+			return true
+		}).Should(BeTrue())
+
+		//Deleting plugin
+		err = test.K8sClient.Delete(ctx, testPlugin)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(func(g Gomega) bool {
+			err = test.K8sClient.List(ctx, pluginList)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(len(pluginList.Items)).To(BeEquivalentTo(0))
+			return true
+		}).Should(BeTrue())
```

Comment:
There is a test helper function that will delete the resource and check that it is eventually cleaned up.
```suggestion
                test.EventuallyDeleted(test.Ctx, test.K8sClient, testPlugin)
```

Reply:
Done